### PR TITLE
feat(evm): add spam transaction caching

### PR DESCRIPTION
**Pull request by moxw**
Add new constant for spam transactions. Implement dynamic caching for last query timestamps to optimize spam detection logic.

Addresses: https://github.com/orgs/rotki/projects/11?pane=issue&itemId=45914187

### CODE REVIEW

File: rotkehlchen/tests/unit/test_chains_aggregator.py [hunk 1]
```diff
@@ -263,3 +265,32 @@ def test_detect_evm_accounts_spam_tx(polygon_pos_manager: 'PolygonPOSManager') -
...
+        '0xC1BeC4618B212441E367ED363540D5D665e8e4F0',
+    ]:
+        get_or_create_evm_token(
+            userdb=gnosis_manager.node_inquirer.database,
+            evm_address=string_to_evm_address(spam_asset),
+            chain_id=ChainID.GNOSIS,
+            protocol=SPAM_PROTOCOL,
+        )
+
+    assert gnosis_manager.transactions.address_has_been_spammed(evm_address) is True
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/test_chains_aggregator.py:296]: I believe this test is incomplete

1. Add a comment saying that evm_address is the address that only has spam transactions
2. Add a comment stating that the addresses are the addresses of all the unknown spam tokens that the address ahs received
3. You should check that the key in the cache is not present before calling `address_has_been_spammed`
4. Check that after calling `address_has_been_spammed` they key has been set
5. Patch requests.get to ensure assert that the arguments respect the new "from_ts" parameter

File: rotkehlchen/chain/evm/constants.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 ZERO_32_BYTES_HEX: Final = '0x' + '0' * 64
 GENESIS_HASH: Final = deserialize_evm_tx_hash(ZERO_32_BYTES_HEX)  # hash for transactions in genesis block # noqa: E501
 EVM_ADDRESS_REGEX: Final = re.compile(r'\b0x[a-fA-F0-9]{40}\b')
+SPAM_TXNS: Final = 'SPAM_TXNS'
```
Comment (yabirgb) [inline rotkehlchen/chain/evm/constants.py:16]: add a comment stating what this constant is. I believe that the abbrevation we use in the code always is

```suggestion
LAST_SPAM_TSX_CACHE: Final = 'SPAM_TXS'
```

File: rotkehlchen/tests/unit/test_chains_aggregator.py [hunk 3]
```diff
@@ -263,3 +270,70 @@ def test_detect_evm_accounts_spam_tx(polygon_pos_manager: 'PolygonPOSManager') -
...
+            location=gnosis_manager.node_inquirer.chain_name,
+            location_name=LAST_SPAM_TXS_CACHE,
+            account_id=evm_address,
+        )
+    assert result is None
+
+    for spam_asset in [
+        '0x8786B9c1a0C676B191284A4Bc7e448321197BB05',
+        '0xC1BeC4618B212441E367ED363540D5D665e8e4F0',
+    ]:  # spam tokens received by the address
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/test_chains_aggregator.py:307]:
```suggestion
    for spam_asset in (
        '0x8786B9c1a0C676B191284A4Bc7e448321197BB05',
        '0xC1BeC4618B212441E367ED363540D5D665e8e4F0',
    ):  # spam tokens received by the address
```

File: rotkehlchen/tests/unit/test_chains_aggregator.py [hunk 4]
```diff
@@ -263,3 +270,70 @@ def test_detect_evm_accounts_spam_tx(polygon_pos_manager: 'PolygonPOSManager') -
...
+            return MockResponse(501, '{"msg": "some error")')
+        return original_get(url)
+
+    with patch('requests.Session.get', side_effect=mock_block_request) as mocked_get:
+        assert gnosis_manager.transactions.address_has_been_spammed(evm_address) is True
+
+        block_number = gnosis_manager.node_inquirer.get_blocknumber_by_time(ts=result, closest='before')  # noqa: E501
+        # verify that the gnosiscan API get's queried with the last recorded blocknumber
+        assert all(f'startBlock={block_number}' in call.args[0] for call in mocked_get.mock_calls), "URL must contain 'startBlock=' and correct block number"  # noqa: E501
+        assert all('endBlock=' in call.args[0] for call in mocked_get.mock_calls), "URL must contain 'endBlock='"  # noqa: E501
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/test_chains_aggregator.py:339]:
```suggestion
    block_number = gnosis_manager.node_inquirer.get_blocknumber_by_time(ts=result, closest='before')  # noqa: E501
    with patch('requests.get') as mocked_get:
        assert gnosis_manager.transactions.address_has_been_spammed(evm_address) is True

        # verify that the gnosiscan API get's queried with the last recorded blocknumber
        assert all(f'startBlock={block_number}' in call.args[0] for call in mocked_get.mock_calls), "URL must contain 'startBlock=' and correct block number"  # noqa: E501
```

File: rotkehlchen/tests/unit/test_chains_aggregator.py [hunk 5]
```diff
@@ -263,3 +270,70 @@ def test_detect_evm_accounts_spam_tx(polygon_pos_manager: 'PolygonPOSManager') -
...
+
+    The tested address has received the following spam tokens:
+    eip155:100/erc20:0x8786B9c1a0C676B191284A4Bc7e448321197BB05
+    eip155:100/erc20:0xC1BeC4618B212441E367ED363540D5D665e8e4F0
+    """
+    # This evm address has only received spam transactions
+    evm_address = string_to_evm_address('0xF5d90Ac6747CB3352F05BF61f48b991ACeaE28eB')
+    database = gnosis_manager.node_inquirer.database
+    original_get = requests.get
+
```
Comment (yabirgb) [inline rotkehlchen/tests/unit/test_chains_aggregator.py:292]:
```suggestion

```

**Review by LefterisJP — APPROVED** [review]
LGTM. Solves https://github.com/orgs/rotki/projects/11?pane=issue&itemId=45914187

**Pull request merged**